docs/proposals-accepted/202304-query-path-tenancy.md

```diff
@@ -0,0 +1,80 @@
+---
+type: proposal
+title: Tenancy awareness in query path
+status: accepted
+owner: douglascamata
+menu: proposals-accepted
+---
+
+## Tenancy awareness in query path
+
+* **Owners:**
+  * [@douglascamata](https://github.com/douglascamata)
+
+* **Related Tickets:**
+  * [Querier: Added native multi-tenancy support](https://github.com/thanos-io/thanos/pull/4141).
+  * [Proposal: Native Multi-tenancy Support](https://github.com/thanos-io/thanos/pull/4055)
+
+* **Other docs:**
+  * [Multi-tenancy in Thanos](https://thanos.io/tip/operating/multi-tenancy.md/)
+
+This design doc proposes to add tenancy awareness in the query path.
+
+## Why
+
+In a multi-tenant environment, it's important to be able to identify which tenants are experiencing issues and configure (e.g. with different limits) each one of them individually and according to their usage of the platform so that the quality of service can be guaranteed to all the tenants.
+
+### Pitfalls of the current solution
+
+The current lack of tenancy awareness in Thanos' query path makes it impossible to investigate issues related to multi-tenancy without the use of external tools or proxies. For example, it's impossible to determine which tenants are experiencing high latency or high error rates.
```

Comment:
This is not fully correct since we can run different queriers for individual tenants. It is worth rephrasing the sentence a bit to indicate that such a setup is more complex and resource inefficient because we cannot pool resources.

Reply:
Sure thing. I'll tackle this together with the other comment you made on the "Goals" section, @fpetkovski, as they are closely related. 

Indeed one can achieve the goals with at least one querier per tenant, but I think only partially. For example, the Query Frontend and Store Gateway metrics will still lack per-tenant visibility, unless you also additional deploy at least one instance of Query Frontend and Store Gateway per tenant.

Reply:
In the end, to have per-tenant visibility and configuration on the query path **today**, you need to deploy a full query path per tenant. It shows how strong the lack of "true" multi-tenancy is.

Reply:
@fpetkovski I tried to incorporate your feedback into the writing. Please, give it a quick read when you can and let me know if it's better.